internal/metrics/dashboards.go

```diff
@@ -48,14 +43,23 @@ func newDashboardConfigMap(collectionType logging.LogCollectionType) *corev1.Con
 	return cm
 }
 
-func ReconcileDashboards(writer client.Writer, reader client.Reader, collection *logging.CollectionSpec) (err error) {
-	collectionType := logging.LogCollectionTypeFluentd
-	if collection != nil {
-		collectionType = collection.Type
+func ReconcileDashboards(writer client.Writer, reader client.Reader) error {
+	var err error
+	cm := newDashboardConfigMap()
+
+	current := &corev1.ConfigMap{}
+	key := client.ObjectKeyFromObject(cm)
+	err = reader.Get(context.TODO(), key, current)
+
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to get %v configmap: %v", key, err)
 	}
-	cm := newDashboardConfigMap(collectionType)
-	if err := reconcile.Configmap(writer, reader, cm, comparators.CompareLabels); err != nil {
-		return err
+
+	// Reconcile dashboard if not found OR hashes are different
+	if (err != nil && errors.IsNotFound(err)) || current.GetLabels()[DashboardHashName] != cm.Labels[DashboardHashName] {
```

Comment:
Doesn't this already happen in reconcile.Configmap?

Reply:
You are right. I thought comparing `ownerrefs` was going to throw it off but nothing owns it.